File: 704. Binary Search/javascript/main.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function (nums, target) {
  let low = 0;
  let high = nums.length - 1;
  while (high >= low) {
    let middle = Math.floor((low + high) / 2);

    if (target == nums[middle]) {
      return target;
    } else if (target > nums[middle]) {
      low = middle + 1;
    } else {
      high = middle - 1;
    }
  }
  return -1;
};

File: 704. Binary Search/python/main.py
from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        high = len(nums)-1
        low = 0
        while (high >= low):
            middle = (high + low) // 2
            if nums[middle] == target:
                return middle
            elif nums[middle] < target:
                low = middle + 1
            else:
                high = middle -1
        return -1